JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeViewModel.kt

```diff
@@ -23,29 +23,74 @@ import com.example.jetnews.R
 import com.example.jetnews.data.Result
 import com.example.jetnews.data.posts.PostsRepository
 import com.example.jetnews.model.Post
+import com.example.jetnews.model.PostsFeed
 import com.example.jetnews.utils.ErrorMessage
 import kotlinx.coroutines.flow.MutableStateFlow
-import kotlinx.coroutines.flow.StateFlow
-import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.SharingStarted
 import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.stateIn
 import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
 import java.util.UUID
 
 /**
  * UI state for the Home screen
  */
-data class HomeUiState(
-    val posts: List<Post> = emptyList(),
+sealed interface HomeUiState {
```

Comment:
I assume this is a pattern for preventing inconsistent UI states. If that's the case, document this thoroughly as this might be confusing for a lot of people if they aren't aware of the context behind it.